.azure/pipelines/ci.yml

```diff
@@ -816,7 +816,8 @@ stages:
             inputs:
               artifactName: Linux_x64_Logs
               itemPattern: '**/Build.binlog'
-        sourceIndexBuildCommand: dir /B /S $(System.ArtifactsDirectory)
+        # Ensure the targeted SDK is available and native projects are built.
+        sourceIndexBuildCommand: $(Build.SourcesDirectory)\eng\build.cmd -buildNative
```

Comment:
/fyi reviewers: This is new in this iteration. Previous internal build failed because source indexing's `BinLogToSln` tool needs to use the SDK described in our global.json. It tries to get that from .dotnet/ in the sources directory instead of the useless dotnet/ folder in `$(Agent.TempDirectory)`.

@alexperovich things _might_ work better if later build steps in eng/common/templates/job/source-index-stage1.yml but I'm not sure. Working around things for now&hellip;

Reply:
The index stuff shouldn't require the sdk described in global json. I will check things.

Reply:
Actually, running the build outside the source-index job may be a bit more fragile than I first realized. The full paths of cs files and referenced dlls are recorded in the binlog, and those must still resolve when the source-index steps process it. Though a failure isn't horrible if it doesn't happen all the time. The website will remain updated with whatever the last successful execution of this job is, so a failure in this job would just delay the appearance of recent changes until the next build.

Reply:
I'm not sure how to explain it other than my thought about the working directory for the last 3 `script` steps in eng/common/templates/job/source-index-stage1.yml.  the error below from the 'Process Binlog into indexable sln' build step is definitely about the sources directory, not the temp directory where the build installed the 3.1.x SDK.

``` text
Unhandled exception. System.IO.DirectoryNotFoundException: Could not find a part of the path 'D:\mnt\vss\_work\1\s\.dotnet\dotnet'.
```

From an earlier binary log, I see `Property AGENT_TEMPDIRECTORY = D:\a\_work\_temp` and `Property BUILD_SOURCESDIRECTORY = D:\a\_work\1\s`.

----

But, now that you mention it, if the indexing isn't resilient to path changes, I need to grab a binary log from a different (Windows) job. Will do that if [20210927.10](https://dev.azure.com/dnceng/internal/_build/results?buildId=1388704&view=results) also fails. That's indexing now.

Reply:
That error is coming from here https://github.com/dotnet/source-indexer/blob/main/src/SourceBrowser/src/BinLogToSln/Program.cs#L171

The indexer reads the compiler invocation from the binlog, then tries to find all the cs files. If its a binlog created on linux, then that could cause that error.

Reply:
Yea, that error makes perfect sense if its a linux binlog. The binlog would have `dotnet csc.dll` in the command line for the CSC task, but the source indexer code is only stripping `dotnet.exe csc.dll` from it before processing the arguments as compiler args. So it ends up treating `dotnet` as a cs file and then tries to process it.

Reply:
Any thoughts @dotnet/aspnet-build❔

Reply:
I don't like 2 or 3 (especially 3), my preference would be 1 or 4.

Reply:
> I can make it work on linux, but I can't promise anything soon. I vote for 4 in the interim, then once I get it working on linux we can switch.

🆗 we're in agreement. I'll make the change later tonight

Reply:
Latest update resolves this conversation&hellip;